fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalTable.java

```diff
@@ -340,12 +352,28 @@ public Map<String, String> getS3Properties() {
         return catalog.getCatalogProperty().getS3Properties();
     }
 
-    public List<Partition> getHivePartitions(ExprNodeGenericFuncDesc hivePartitionPredicate) throws DdlException {
-        List<Partition> hivePartitions = Lists.newArrayList();
-        PooledHiveMetaStoreClient client = ((HMSExternalCatalog) catalog).getClient();
-        client.listPartitionsByExpr(remoteTable.getDbName(), remoteTable.getTableName(),
-                SerializationUtilities.serializeExpressionToKryo(hivePartitionPredicate), hivePartitions);
-        return hivePartitions;
+    private void initPartitionColumns() {
+        if (partitionColumns != null) {
+            return;
+        }
+        synchronized (this) {
+            if (partitionColumns != null) {
+                return;
+            }
+            List<String> partitionKeys = remoteTable.getPartitionKeys().stream().map(FieldSchema::getName)
+                    .collect(Collectors.toList());
```

Comment:
collect as set to avoid traversing `partitionKeys`

Reply:
done